x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/transform/script/WatcherTransformScript.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.watcher.transform.script;
+
+import org.elasticsearch.script.ParameterMap;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.xpack.core.watcher.execution.WatchExecutionContext;
+import org.elasticsearch.xpack.core.watcher.watch.Payload;
+import org.elasticsearch.xpack.watcher.support.Variables;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A script to transform the results of a watch execution.
+ */
+public abstract class WatcherTransformScript {
+    public static final String[] PARAMETERS = {};
+
+    private static final Map<String, String> DEPRECATIONS;
+
+    static {
+        Map<String, String> deprecations = new HashMap<>();
+        deprecations.put(
+            "ctx",
+            "Accessing variable [ctx] via [params.ctx] from within a watcher_transform script " +
```

Comment:
is it worth testing this somewhere or did I miss the test? Same applies for the `WatcherConditionScript`

Reply:
I added tests in e1e0490